### PR TITLE
[BEAM-390] Remove ParDo.named

### PR DESCRIPTION
**Pull request by bjchambers**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [*] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [*] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [*] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @davorbonaci @dhalperi

**Comment by dhalperi** [comment]
Doesn't build.

**Comment by bjchambers** [comment]
Should build now. I missed some Javadocs and a unique name.

File: examples/java8/src/main/java/org/apache/beam/examples/complete/game/utils/WriteToBigQuery.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ protected TableSchema getSchema() {
   @Override
   public PDone apply(PCollection<T> teamAndScore) {
     return teamAndScore
-      .apply(ParDo.named("ConvertToRow").of(new BuildRowFn()))
+      .apply("ConvertToRows", ParDo.of(new BuildRowFn()))
```
Comment (dhalperi) [inline examples/java8/src/main/java/org/apache/beam/examples/complete/game/utils/WriteToBigQuery.java]: extra `s` added

File: examples/java8/src/main/java/org/apache/beam/examples/complete/game/utils/WriteWindowedToBigQuery.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public void processElement(ProcessContext c) {
   @Override
   public PDone apply(PCollection<T> teamAndScore) {
     return teamAndScore
-      .apply(ParDo.named("ConvertToRow").of(new BuildRowFn()))
+      .apply("ConvertToRow", ParDo.of(new BuildRowFn()))
```
Comment (dhalperi) [inline examples/java8/src/main/java/org/apache/beam/examples/complete/game/utils/WriteWindowedToBigQuery.java]: (fine with `rows` or `row` but be consistent across here and above)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 3]
```diff
@@ -647,8 +647,8 @@ protected String getKindString() {
 
   /////////////////////////////////////////////////////////////////////////////
 
-  private static <T> PTransform<PCollection<? extends T>, PCollection<T>> identity() {
-    return ParDo.named("Identity").of(new DoFn<T, T>() {
+  private static <T> PTransform<PCollection<? extends T>, PCollection<T>> identityDoFn() {
+    return ParDo.of(new DoFn<T, T>() {
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: still returns a PTransform. Is that intended?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/Reshuffle.java [hunk 4]
```diff
@@ -69,7 +70,7 @@ private Reshuffle() {
         // Set the windowing strategy directly, so that it doesn't get counted as the user having
         // set allowed lateness.
         .setWindowingStrategyInternal(originalStrategy)
-        .apply(ParDo.named("ExpandIterable").of(
+        .apply("ExpandIterabel", ParDo.of(
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/Reshuffle.java]: spell-o

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/ValueWithRecordId.java [hunk 5]
```diff
@@ -142,15 +139,11 @@ public void verifyDeterministic() throws NonDeterministicException {
...
-    return ParDo.named("StripIds")
-        .of(
-            new DoFn<ValueWithRecordId<T>, T>() {
-              @Override
-              public void processElement(ProcessContext c) {
-                c.output(c.element().getValue());
-              }
-            });
+  /** {@link DoFn} to turn a {@code ValueWithRecordId<T>} back to the value {@code T}. */
+  public static class StripIds<T> extends DoFn<ValueWithRecordId<T>, T> {
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/ValueWithRecordId.java]: StripIdsDoFn?

Alternately, keep it a PTransform?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java [hunk 6]
```diff
@@ -815,37 +816,22 @@ public void testParDoGetName() {
...
+      PCollection<String> output4 = input.apply("TestDoFn", ParDo.of(new TestDoFn()));
       assertEquals("TestDoFn.out", output4.getName());
     }
 
     {
-      PCollection<String> output5 =
-          input
-              .apply(ParDo.of(new StrangelyNamedDoer()));
+      PCollection<String> output5 = input.apply(ParDo.of(new StrangelyNamedDoer()));
       assertEquals("ParDo(StrangelyNamedDoer).out",
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java]: never seen this code before, but lol.
No action required.

**Comment by dhalperi** [comment]
LGTM. please self-merge after addressing any comments.

**Comment by davorbonaci** [comment]
LGTM

**Pull request closed without merging**